testing/dart/fragment_shader_test.dart

```diff
@@ -332,6 +332,38 @@ void main() async {
     shader.dispose();
   });
 
+  // This test can't rely on actual pixels rendered since it needs to run on a
+  // metal shader on iOS. instead parse the source code.
+  test('impellerc orders samplers in metal shader according to declaration and not usage', () async {
+    final Directory directory = shaderDirectory('iplr-remap');
+    final String data = File(path.join(directory.path, 'shader_with_samplers.frag.iplr')).readAsStringSync();
```

Comment:
The `.iplr` file is flatbuffer encoded.

Reply:
When generating an `.iplr`, the plain text platform specific shader might also be generated nearby. Sorry, I'm blanking on the details, but `impellerc` should be able to generate both at the same time, somehow. Then you can check the bytes of the plain text shader.